[Bug]: GSD's auto-mode is entering a stuck loop, repeatedly dispatching "research-milestone" and "research-slice" units

### Bug type

Regression (worked before, now fails)

### Summary

│  ## Problem
│  GSD's auto-mode is entering a stuck loop, repeatedly dispatching "research-milestone" and "research-slice" units
│  without successful verification. Additionally, the Doctor reports multiple active requirements with no primary
│  owning slice.
│
│  ## Root Cause
│  While direct source code access was unavailable, forensic analysis suggests two primary root causes:
│
│  1.  **Faulty Unit Verification/State Update:** In `auto-unit-closeout.ts` or `auto-verification.ts`, there's likely
│  a bug preventing units from being definitively marked as completed after execution. This could be due to
│  insufficient artifact checks, timing issues, or non-atomic state updates to `completed-units.json`.
│      *   **Evidence:** "stuck-loop" anomalies for `research-slice/M001/parallel-research` (dispatched 2 times) and
│  `research-milestone/M004` (dispatched 7 times), and "Completed Keys: 0" in the metrics summary.
│
│  2.  **Incomplete Requirement-Slice Mapping Persistence:** The planning phase is failing to correctly associate
│  active requirements with their owning slices and persist this information in `.gsd/REQUIREMENTS.md`. This likely
│  points to an issue in how planning commands update GSD's internal state and write it to disk (e.g., `commands.ts`,
│  `json-persistence.ts`, `files.ts`).
│      *   **Evidence:** Multiple "doctor-issue" errors for `active_requirement_missing_owner` (e.g., R001, R002,
│  R003, etc.).
│
│  ## Expected Behavior
│
│  1.  Units should be dispatched, executed, and, upon successful completion and artifact verification, marked as
│  complete without repeated re-dispatching.
│  2.  Active requirements should be correctly mapped to primary owning slices during the planning phase and this
│  mapping should be persisted.
│
│  ## Concrete Fix Suggestions
│
│  **For stuck loops:**
│  *   **Improve idempotency:** Ensure unit handlers are idempotent.
│  *   **Robust artifact verification:** Enhance `auto-verification.ts` to perform more thorough checks on expected
│  artifacts.
│  *   **Atomic state updates:** Update `completed-units.json` and other state files atomically after all verification
│  steps.
│  *   **Detailed Logging:** Add verbose logging in `auto-unit-closeout.ts` and `auto-verification.ts` to debug
│  verification failures.
│
│  **For `active_requirement_missing_owner`:**
│  *   **Enforce mapping:** Ensure all active requirements are explicitly assigned a primary owning slice during
│  `plan-milestone` and `plan-slice`.
│  *   **Review Doctor check:** Verify the `active_requirement_missing_owner` check in `doctor-checks.ts` for
│  accuracy.
│
│  ## Environment
│  - GSD version: 2.75.0
│  - Model: gemini-3-flash (for research-milestone), gemma-3-27b-it (for plan-milestone)
│  - Unit: research-milestone/M004, research-slice/M001/parallel-research
│
│  ## Reproduction Context
│  This occurred during an auto-mode session, specifically involving research and planning units. The exact user
│  command that triggered the auto-mode was `/gsd setup llm`, but the issues are internal to GSD's state management
│  and execution loop.
│
│  ## Forensic Evidence
│  - **[WARNING] stuck-loop**: Unit research-slice/M001/parallel-research was dispatched 2 times
│  - **[ERROR] stuck-loop**: Unit research-milestone/M004 was dispatched 7 times
│  - **[ERROR] doctor-issue**: Doctor: R001 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R002 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R003 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R004 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R005 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R006 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R008 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R009 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R013 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R014 is Active but has no primary owning slice
│  - **[ERROR] doctor-issue**: Doctor: R015 is Active but has no primary owning slice
│
│  ---
│  *Auto-generated by `/gsd forensics`*
│  GSD_ISSUE_BODY
│
│  ISSUE_URL=$(gh issue create --repo gsd-build/gsd-2 \
│    --title "BUG: Auto-mode stuck loop and unassigned active requirements" \
│    --label "auto-generated" \
│    --body-file /tmp/gsd-forensic-issue.md)
│  rm -f /tmp/gsd-forensic-issue.md
│
│  ISSUE_NUM=$(echo "$ISSUE_URL" | grep -oE '[0-9]+$')
│  ISSUE_ID=$(gh api graphql -f query='{ repository(owner:"gsd-build",name:"gsd-2") { issue(number:'"$ISSUE_NUM"') {
│  id } } }' --jq '.data.repository.issue.id')
│  TYPE_ID=$(gh api graphql -f query='{ repository(owner:"gsd-build",name:"gsd-2") { issueTypes(first:20) { nodes { id
│  name } } } }' --jq '.data.repository.issueTypes.nodes[] | select(.name=="Bug") | .id')
│  gh api graphql -f query='mutation { updateIssue(input:{id:"'"$ISSUE_ID"'",issueTypeId:"'"$TYPE_ID"'"}) { issue {
│  number } } }'
│
│
│  ... (2 earlier lines, ctrl+o to expand)
│  /usr/bin/bash: line 67: gh: command not found
│  /usr/bin/bash: line 68: gh: command not found
│
│
│  Command exited with code 127

─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
• GSD - gemini-2.5-flash                                                                                 2026-04-17 19:03
│  The gh command is still not found, preventing the creation of the GitHub issue. Despite the earlier winget install
│  GitHub.cli command showing successful installation, it appears the gh executable is not in the shell's PATH.
│
│  Since I cannot directly interact with the file system to debug the PATH issue or inspect GSD's source code, I will
│  provide the issue details directly in the chat for the user to file manually.
│
│  The full forensic report was saved locally, and I've analyzed the issues in detail.
│
│  What happened:
│  GSD's auto-mode entered a stuck loop, repeatedly dispatching "research-milestone" and "research-slice" units
│  without successful verification. This is evidenced by research-milestone/M004 being dispatched 7 times and
│  research-slice/M001/parallel-research twice. Simultaneously, the Doctor reported numerous active requirements
│  (R001-R006, R008, R009, R013-R015) that are active but lack a primary owning slice in .gsd/REQUIREMENTS.md. This
│  indicates a breakdown in both task completion verification and planning state management within GSD.
│
│  Why it happened (Root Cause Hypothesis):
│  1. Faulty Unit Verification/State Update: The primary cause of the stuck loops is likely a bug in GSD's auto-mode
│  engine, specifically in the unit closeout or verification logic (e.g., auto-unit-closeout.ts or
│  auto-verification.ts). It appears that units, even if they complete their actions, are not being robustly marked as
│  "done." This could be due to:
│      - Insufficient checks against expected artifacts (e.g., the presence or content of M###-RESEARCH.md or
│  M###-ROADMAP.md).
│      - Timing issues or race conditions where the verification logic runs before artifacts are fully written to
│  disk.
│      - Non-atomic updates to the completed-units.json ledger, causing the system to believe a unit is still
│  incomplete.
│      - The "Completed Keys: 0" in the metrics summary strongly supports a verification failure.
│  2. Incomplete Requirement-Slice Mapping Persistence: The "active_requirement_missing_owner" errors suggest a
│  separate, but related, issue in the planning phase. Active requirements are not being correctly associated with
│  their owning slices and this mapping is not being persisted to .gsd/REQUIREMENTS.md. This might be due to a bug in
│  the planning commands (commands.ts) or the general state persistence mechanisms (json-persistence.ts, files.ts).
│
│  Recovery for the User:
│
│  To get unstuck, the user would likely need to manually intervene.
│  1. Clear GSD's internal state: Manually delete .gsd/auto.lock and potentially .gsd/STATE.md to force a clean
│  restart of the auto-mode. This might lose some in-progress task state if not carefully managed.
│  2. Manually update completed-units.json: If the user can confirm that the research-milestone/M004 and
│  research-slice/M001/parallel-research units did produce their expected artifacts, they could try to manually update
│  .gsd/completed-units.json to reflect their completion, but this is risky and prone to errors.
│
│  Here are the details for a GitHub issue:
│
│  ```
│    ## Problem
│    GSD's auto-mode is entering a stuck loop, repeatedly dispatching "research-milestone" and "research-slice" units
│  without successful verification. Additionally, the Doctor reports multiple active requirements with no primary
│  owning slice.
│
│    ## Root Cause
│    While direct source code access was unavailable, forensic analysis suggests two primary root causes:
│
│    1.  **Faulty Unit Verification/State Update:** In `auto-unit-closeout.ts` or `auto-verification.ts`, there's
│  likely a bug preventing units from being definitively marked as completed after execution. This could be due to
│  insufficient artifact checks, timing issues, or non-atomic state updates to `completed-units.json`.
│        *   **Evidence:** "stuck-loop" anomalies for `research-slice/M001/parallel-research` (dispatched 2 times) and
│  `research-milestone/M004` (dispatched 7 times), and "Completed Keys: 0" in the metrics summary.
│
│    2.  **Incomplete Requirement-Slice Mapping Persistence:** The planning phase is failing to correctly associate
│  active requirements with their owning slices and persist this information in `.gsd/REQUIREMENTS.md`. This likely
│  points to an issue in how planning commands update GSD's internal state and write it to disk (e.g., `commands.ts`,
│  `json-persistence.ts`, `files.ts`).
│        *   **Evidence:** Multiple "doctor-issue" errors for `active_requirement_missing_owner` (e.g., R001, R002,
│  R003, etc.).
│
│    ## Expected Behavior
│
│    1.  Units should be dispatched, executed, and, upon successful completion and artifact verification, marked as
│  complete without repeated re-dispatching.
│    2.  Active requirements should be correctly mapped to primary owning slices during the planning phase and this
│  mapping should be persisted.
│
│    ## Concrete Fix Suggestions
│
│    **For stuck loops:**
│    *   **Improve idempotency:** Ensure unit handlers are idempotent.
│    *   **Robust artifact verification:** Enhance `auto-verification.ts` to perform more thorough checks on expected
│  artifacts.
│    *   **Atomic state updates:** Update `completed-units.json` and other state files atomically after all
│  verification steps.
│    *   **Detailed Logging:** Add verbose logging in `auto-unit-closeout.ts` and `auto-verification.ts` to debug
│  verification failures.
│
│    **For `active_requirement_missing_owner`:**
│    *   **Enforce mapping:** Ensure all active requirements are explicitly assigned a primary owning slice during
│  `plan-milestone` and `plan-slice`.
│    *   **Review Doctor check:** Verify the `active_requirement_missing_owner` check in `doctor-checks.ts` for
│  accuracy.
│
│    ## Environment
│    - GSD version: 2.75.0
│    - Model: gemini-3-flash (for research-milestone), gemma-3-27b-it (for plan-milestone)
│    - Unit: research-milestone/M004, research-slice/M001/parallel-research
│
│    ## Reproduction Context
│    This occurred during an auto-mode session, specifically involving research and planning units. The exact user
│  command that triggered the auto-mode was `/gsd setup llm`, but the issues are internal to GSD's state management
│  and execution loop.
│
│    ## Forensic Evidence
│    - **[WARNING] stuck-loop**: Unit research-slice/M001/parallel-research was dispatched 2 times
│    - **[ERROR] stuck-loop**: Unit research-milestone/M004 was dispatched 7 times
│    - **[ERROR] doctor-issue**: Doctor: R001 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R002 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R003 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R004 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R005 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R006 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R008 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R009 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R013 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R014 is Active but has no primary owning slice
│    - **[ERROR] doctor-issue**: Doctor: R015 is Active but has no primary owning slice
│
│    ---
│    *Auto-generated by `/gsd forensics`*

### Steps to reproduce

1. Setup milestone
2. Execute
3. Error: auto-execute stuck

### Expected behavior

GSD should not get stuck

### Actual behavior

GSD auto-execute stuck in a loop

### GSD version

v2.75

### Node.js version

24.13

### Operating system

Windows 11

### Install method

npm global install

### Affected area

Auto-mode / dispatch loop

### Model and provider

_No response_

### Logs, screenshots, and evidence

```shell

```

### Impact

Severity: Major when it occurs
Frequency: To be determined

### Additional information

_No response_

--- a/src/resources/extensions/gsd/auto-artifact-paths.ts
+++ b/src/resources/extensions/gsd/auto-artifact-paths.ts
@@ -43,6 +43,16 @@ export function resolveExpectedArtifactPath(
       return dir ? join(dir, buildMilestoneFileName(mid, "ROADMAP")) : null;
     }
     case "research-slice": {
+      // #4414: Sentinel unitId "{mid}/parallel-research" fans out across
+      // multiple slices. Resolve to a milestone-level placeholder path so
+      // blocker escalation has somewhere to write. Verification for this
+      // sentinel is handled directly in verifyExpectedArtifact.
+      if (sid === "parallel-research") {
+        const mdir = resolveMilestonePath(base, mid);
+        return mdir
+          ? join(mdir, buildMilestoneFileName(mid, "PARALLEL-BLOCKER"))
+          : null;
+      }
       const dir = resolveSlicePath(base, mid, sid!);
       return dir ? join(dir, buildSliceFileName(sid!, "RESEARCH")) : null;
     }
@@ -114,6 +124,9 @@ export function diagnoseExpectedArtifact(
     case "plan-milestone":
       return `${relMilestoneFile(base, mid, "ROADMAP")} (milestone roadmap)`;
     case "research-slice":
+      if (sid === "parallel-research") {
+        return `${relMilestoneFile(base, mid, "PARALLEL-BLOCKER")} (parallel slice research sentinel)`;
+      }
       return `${relSliceFile(base, mid, sid!, "RESEARCH")} (slice research)`;
     case "plan-slice":
       return `${relSliceFile(base, mid, sid!, "PLAN")} (slice plan)`;

--- a/src/resources/extensions/gsd/auto-dispatch.ts
+++ b/src/resources/extensions/gsd/auto-dispatch.ts
@@ -452,6 +452,12 @@ export const DISPATCH_RULES: DispatchRule[] = [
       // Only dispatch parallel if 2+ slices are ready
       if (researchReadySlices.length < 2) return null;
 
+      // #4414: If a previous parallel-research attempt escalated to a blocker
+      // placeholder, skip this rule and fall through to per-slice research
+      // (or other rules) rather than re-dispatching the same failing unit.
+      const parallelBlocker = resolveMilestoneFile(basePath, mid, "PARALLEL-BLOCKER");
+      if (parallelBlocker) return null;
+
       return {
         action: "dispatch",
         unitType: "research-slice",

--- a/src/resources/extensions/gsd/auto-recovery.ts
+++ b/src/resources/extensions/gsd/auto-recovery.ts
@@ -261,6 +261,45 @@ export function verifyExpectedArtifact(
     return true;
   }
 
+  // #4414: research-slice parallel-research sentinel. The unitId
+  // `{mid}/parallel-research` is not a real slice — it triggers a single agent
+  // that fans out research across multiple slices. Verify success by checking
+  // that every slice which was "research-ready" in the roadmap now has a
+  // RESEARCH file. Without this, resolveExpectedArtifactPath returns null and
+  // the retry/escalation machinery silently re-dispatches forever.
+  //
+  // NOTE: this predicate mirrors the dispatch rule at
+  // auto-dispatch.ts parallel-research-slices — keep the two in sync.
+  if (unitType === "research-slice" && unitId.endsWith("/parallel-research")) {
+    const { milestone: mid } = parseUnitId(unitId);
+    if (!mid) return false;
+    const roadmapFile = resolveMilestoneFile(base, mid, "ROADMAP");
+    if (!roadmapFile || !existsSync(roadmapFile)) {
+      logWarning("recovery", `verify-fail ${unitType} ${unitId}: roadmap missing`);
+      return false;
+    }
+    try {
+      const roadmap = parseLegacyRoadmap(readFileSync(roadmapFile, "utf-8"));
+      const milestoneResearchFile = resolveMilestoneFile(base, mid, "RESEARCH");
+      for (const slice of roadmap.slices) {
+        if (slice.done) continue;
+        if (milestoneResearchFile && slice.id === "S01") continue;
+        const depsComplete = (slice.depends ?? []).every((depId) =>
+          !!resolveSliceFile(base, mid, depId, "SUMMARY"),
+        );
+        if (!depsComplete) continue;
+        if (!resolveSliceFile(base, mid, slice.id, "RESEARCH")) {
+          logWarning("recovery", `verify-fail ${unitType} ${unitId}: slice ${slice.id} missing RESEARCH`);
+          return false;
+        }
+      }
+      return true;
+    } catch (err) {
+      logWarning("recovery", `parallel-research verification failed: ${err instanceof Error ? err.message : String(err)}`);
+      return false;
+    }
+  }
+
   const absPath = resolveExpectedArtifactPath(unitType, unitId, base);
   // For unit types with no verifiable artifact (null path), the parent directory
   // is missing on disk — treat as stale completion state so the key gets evicted (#313).
@@ -454,6 +493,14 @@ export function writeBlockerPlaceholder(
     `Review and replace this file before relying on downstream artifacts.`,
   ].join("\n");
   writeFileSync(absPath, content, "utf-8");
+
+  // #4414: Clear caches so subsequent dispatch guards (e.g.
+  // resolveMilestoneFile) see the placeholder file. Without this, the
+  // cached directory listing is stale and the dispatch rule re-fires,
+  // producing an infinite loop despite the placeholder being on disk.
+  // Matches the pattern used in verifyExpectedArtifact above.
+  clearPathCache();
+  clearParseCache();
 
   // Mark the task/slice as complete in the DB so verifyExpectedArtifact passes.
   // Without this, the DB status stays "pending" and the dispatch loop

--- a/src/resources/extensions/gsd/doctor.ts
+++ b/src/resources/extensions/gsd/doctor.ts
@@ -172,8 +172,14 @@ function auditRequirements(content: string | null): DoctorIssue[] {
     const notes = block.match(/^-\s+Notes:\s+(.+)$/m)?.[1]?.trim().toLowerCase() ?? "";
 
     if (status === "active" && (!owner || owner === "none" || owner === "none yet")) {
+      // #4414: Downgrade to warning. A newly-created requirement has
+      // primary_owner='' by default until the planning agent wires it to
+      // a slice via gsd_requirement_update. Flagging this as an error
+      // during normal planning is noisy — the real failure mode is when
+      // it persists past milestone completion, which is covered by other
+      // audits. Keep the signal but don't treat it as a blocker.
       issues.push({
-        severity: "error",
+        severity: "warning",
         code: "active_requirement_missing_owner",
         scope: "project",
         unitId: requirementId,

--- a/src/resources/extensions/gsd/tests/auto-recovery.test.ts
+++ b/src/resources/extensions/gsd/tests/auto-recovery.test.ts
@@ -5,7 +5,8 @@ import { join } from "node:path";
 import { tmpdir } from "node:os";
 import { randomUUID } from "node:crypto";
 
-import { verifyExpectedArtifact, hasImplementationArtifacts, resolveExpectedArtifactPath, diagnoseExpectedArtifact, buildLoopRemediationSteps } from "../auto-recovery.ts";
+import { verifyExpectedArtifact, hasImplementationArtifacts, resolveExpectedArtifactPath, diagnoseExpectedArtifact, buildLoopRemediationSteps, writeBlockerPlaceholder } from "../auto-recovery.ts";
+import { resolveMilestoneFile } from "../paths.ts";
 import { openDatabase, closeDatabase, insertMilestone, insertSlice, insertGateRow } from "../gsd-db.ts";
 import { clearParseCache } from "../files.ts";
 import { parseRoadmap } from "../parsers-legacy.ts";
@@ -711,4 +712,125 @@ test("verifyExpectedArtifact checks pending gate-evaluate artifacts without ESM 
   const verified = verifyExpectedArtifact("gate-evaluate", "M001/S01/gates+Q3", base);
 
   assert.equal(verified, false, "pending gates should keep gate-evaluate unverified");
+});
+
+// ─── #4414 regressions ────────────────────────────────────────────────────────
+
+test("#4414: writeBlockerPlaceholder invalidates path cache so dispatch guard sees file", () => {
+  const base = makeTmpBase();
+  try {
+    // Prime the readdir cache by resolving a DIFFERENT file first — this
+    // mirrors the stuck-loop condition where the dispatch guard cached an
+    // empty directory listing before the placeholder was written.
+    invalidateAllCaches();
+    assert.equal(
+      resolveMilestoneFile(base, "M001", "RESEARCH"),
+      null,
+      "no RESEARCH file yet",
+    );
+
+    const result = writeBlockerPlaceholder(
+      "research-milestone",
+      "M001",
+      base,
+      "verification retries exhausted",
+    );
+    assert.ok(result, "placeholder path returned");
+
+    // After writeBlockerPlaceholder, the dispatch guard must see the new file
+    // immediately — otherwise the rule re-fires (#4414, 7× re-dispatch).
+    const postResolve = resolveMilestoneFile(base, "M001", "RESEARCH");
+    assert.ok(
+      postResolve,
+      "resolveMilestoneFile finds the placeholder post-write (cache invalidated)",
+    );
+  } finally {
+    cleanup(base);
+  }
+});
+
+test("#4414: parallel-research sentinel path does not collide with RESEARCH suffix", () => {
+  const base = makeTmpBase();
+  try {
+    // Write only the parallel-research blocker (sentinel).
+    const sentinel = resolveExpectedArtifactPath(
+      "research-slice",
+      "M001/parallel-research",
+      base,
+    );
+    assert.ok(sentinel, "sentinel path resolves for parallel-research");
+    writeFileSync(sentinel!, "# blocker\n", "utf-8");
+
+    // Critical: the sentinel filename must NOT be matched by the legacy regex
+    // used when callers look up milestone-level RESEARCH. Otherwise the
+    // dispatch guard for research-milestone would short-circuit falsely.
+    const milestoneResearch = resolveMilestoneFile(base, "M001", "RESEARCH");
+    assert.equal(
+      milestoneResearch,
+      null,
+      "sentinel must not be mistaken for M001-RESEARCH.md via legacy pattern match",
+    );
+  } finally {
+    cleanup(base);
+  }
+});
+
+test("#4414: verifyExpectedArtifact parallel-research succeeds when all research-ready slices have RESEARCH", () => {
+  const base = makeTmpBase();
+  try {
+    mkdirSync(join(base, ".gsd", "milestones", "M001", "slices", "S02", "tasks"), { recursive: true });
+    mkdirSync(join(base, ".gsd", "milestones", "M001", "slices", "S03", "tasks"), { recursive: true });
+
+    // Minimal roadmap with three slices
+    writeFileSync(
+      join(base, ".gsd", "milestones", "M001", "M001-ROADMAP.md"),
+      [
+        "# M001: Regression",
+        "",
+        "## Slices",
+        "",
+        "- [ ] **S01: Alpha** `risk:low` `depends:[]`",
+        "- [ ] **S02: Beta** `risk:low` `depends:[]`",
+        "- [ ] **S03: Gamma** `risk:low` `depends:[]`",
+        "",
+      ].join("\n"),
+      "utf-8",
+    );
+
+    // Only 2 of 3 have RESEARCH — should fail verification
+    writeFileSync(
+      join(base, ".gsd", "milestones", "M001", "slices", "S01", "S01-RESEARCH.md"),
+      "# research",
+      "utf-8",
+    );
+    writeFileSync(
+      join(base, ".gsd", "milestones", "M001", "slices", "S02", "S02-RESEARCH.md"),
+      "# research",
+      "utf-8",
+    );
+
+    clearParseCache();
+    invalidateAllCaches();
+    assert.equal(
+      verifyExpectedArtifact("research-slice", "M001/parallel-research", base),
+      false,
+      "missing S03 RESEARCH → verification fails",
+    );
+
+    // All three RESEARCH present → verification passes
+    writeFileSync(
+      join(base, ".gsd", "milestones", "M001", "slices", "S03", "S03-RESEARCH.md"),
+      "# research",
+      "utf-8",
+    );
+    clearParseCache();
+    invalidateAllCaches();
+    assert.equal(
+      verifyExpectedArtifact("research-slice", "M001/parallel-research", base),
+      true,
+      "all slices have RESEARCH → verification passes",
+    );
+  } finally {
+    cleanup(base);
+  }
 });

--- a/src/resources/extensions/gsd/tests/requirements.test.ts
+++ b/src/resources/extensions/gsd/tests/requirements.test.ts
@@ -95,6 +95,15 @@ describe('requirements', () => {
     assert.ok(report.issues.some(issue => issue.code === "active_requirement_missing_owner"), "doctor flags missing owner");
   });
 
+  // #4414: active_requirement_missing_owner is a planning-hygiene signal,
+  // not a correctness blocker — severity must be warning, not error.
+  test('#4414: active_requirement_missing_owner is a warning, not an error', async () => {
+    const report = await runGSDDoctor(base);
+    const issue = report.issues.find(i => i.code === "active_requirement_missing_owner");
+    assert.ok(issue, "issue is present");
+    assert.equal(issue!.severity, "warning", "severity downgraded so doctor report.ok is not flipped to false");
+  });
+
   after(() => {
     rmSync(base, { recursive: true, force: true });
   });

